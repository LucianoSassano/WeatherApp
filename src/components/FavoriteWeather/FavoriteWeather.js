import React, { useState } from "react";
import "../FavoriteWeather/FavoriteWeather.css";

const FavoriteWeather = (props) => {
  const [location, setLocation] = useState(props);
 

  const handleDelete = () => {
    const localStorageFavs = JSON.parse(localStorage.getItem("favorites"));
    let jsonFavs = [];
    if (Array.isArray(localStorageFavs)) jsonFavs = localStorageFavs;

    jsonFavs = jsonFavs.filter(
      (weather) =>
        weather.city !== props.city && weather.country !== props.country
    );
    console.log(jsonFavs);
    localStorage.setItem("favorites", JSON.stringify(jsonFavs));
    setLocation(jsonFavs);
  };

  return (
    <>
      <div data-testid="favoriteWeather" className="weather-container">
        {location.city ? (
          <div className="favorites-data" aria-label="weather-result">
            <h1
              aria-label="city name and country of city"
              value={(location.city, location.country)}
            >
              {location.city},{location.country}
            </h1>
            <h3
              aria-label="location temperature in celcius degrees"
              value={location.temp}
            >
              {location.temp + "Â°C"}
            </h3>
            <h2
              aria-label="location weather description"
              value={location.description}
            >
              {location.description}
            </h2>
            <h4>{location.detail}</h4>
            <img
              src={`https://openweathermap.org/img/wn/${location.icon}@2x.png`}
            ></img>
            <br></br>
            <button
              type="button"
              className="btn btn-danger"
              onClick={handleDelete}
              aria-label="delete forecaste from favorites"
            >
              Delete
            </button>
          </div>
        ) : (
          ""
        )}
      </div>
    </>
  );
};

export default FavoriteWeather;
