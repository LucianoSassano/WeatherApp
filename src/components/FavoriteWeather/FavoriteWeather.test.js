import React from 'react';
import {cleanup} from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import FavoriteWeather from './FavoriteWeather.js';


afterEach(cleanup);

it('FavoriteWeather renders correctly', () => {

    const div = document.createElement("div");
    ReactDOM.render(<FavoriteWeather></FavoriteWeather>, div)
})


it("Weather component matches snapshot", () => {

    const favoriteWeather = renderer.create(<FavoriteWeather></FavoriteWeather>).toJSON();
    expect(favoriteWeather).toMatchSnapshot();

})