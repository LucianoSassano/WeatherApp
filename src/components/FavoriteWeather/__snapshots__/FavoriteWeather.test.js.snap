// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component matches snapshot 1`] = `
<div
  className="weather-container"
  data-testid="favoriteWeather"
>
  
</div>
`;
